src/core/Solana/SolanaStepExecutor.ts

```diff
@@ -37,6 +37,16 @@ export class SolanaStepExecutor extends BaseStepExecutor {
     this.walletAdapter = options.walletAdapter
   }
 
+  checkWalletAdapter = (step: LiFiStepExtended) => {
+    // Prevent execution of the quote by wallet different from the one which requested the quote
+    if (this.walletAdapter.publicKey!.toString() !== step.action.fromAddress) {
+      throw new TransactionError(
+        LiFiErrorCode.WalletChangedDuringExecution,
+        'The wallet that requested the quote does not match the wallet attempting to sign the transaction.'
```

Comment:
Should this be worded the same as the error message for the EVMStepExecutor?

Reply:
Oh, yes, thank you 😄